docs: clarify asset handling (#4128)
webpack/webpack.js.org: src/content/guides/asset-modules.md

- `asset/source` exports the source code of the asset. Previously achievable by using `raw-loader`.
- `asset` automatically chooses between exporting a data URI and emitting a separate file. Previously achievable by using `url-loader` with asset size limit.

> When using the old assets loaders (i.e. `file-loader`/`url-loader`/`raw-loader`) in webpack 5 you might want to exclude them from the assets module processing to prevent asset duplication. This can be done by overriding the asset configuration for that asset type by setting it's module type to 'javascript/auto'.

Reviewer: ```suggestion
When using the old assets loaders (i.e. `file-loader`/`url-loader`/`raw-loader`) along with Asset Module in webpack 5, you might want to stop Asset Module from processing your assets as that would result in asset duplication. This can be done by setting asset's module type to `'javascript/auto'`.
```

``` diff
module.exports = {
  module: {
   rules: [
      {
        test: /\.(png|jpg|gif)$/i,
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 8192,
            }
          },
        ],
      },
+     {
+       test: /\.png/,
+       type: 'javascript/auto'
+     }
   ]
  },
}
```
To exclude assets that came from new URL calls from the asset loaders add `dependency: { not: 'url' }` to the loader configuration.
``` diff
module.exports = {
  module: {
    rules: [
      {

>         test: /\.(png|jpg|gif)$/i,

Reviewer: ```suggestion
        test: /\.(png|jpg|gif)$/i,
+       dependency: { not: ['url'] }, 
```
The `dependency` needs to be set besides `test` etc.

- `asset/source` exports the source code of the asset. Previously achievable by using `raw-loader`.
- `asset` automatically chooses between exporting a data URI and emitting a separate file. Previously achievable by using `url-loader` with asset size limit.

> W>When using the old assets loaders in webpack 5 you might want to exclude them from the assets module processing to prevent asset duplication. This can be done by overriding the asset configuration for that asset type by setting it's module type to "javascript/auto". To exclude assets that came from new URL calls from the asset loaders add dependency: { not: "url" } to the loader configuration.

Reviewer 2: Let's clarify what is `old assets loaders` - i.e. `file-loader`/`url-loader`/`raw-loader` and add simple examples for `javascript/auto` and `dependency: { not: "url" }`

Author: I added examples and removed the W> because I can't see how it will rendered.
